catatonit.c

```diff
@@ -193,20 +194,46 @@ static int make_foreground(sigset_t *sigmask)
 	return 0;
 }
 
+/*
+ * If the LISTEN_PID environment variable is set to "1", rewrite it to
+ * point to the child process.
+ */
+static void rewrite_listen_pid_env(int parent_pid)
+{
+	char *listen_pid;
+	char pid_str[32];
+
+	sprintf(pid_str, "%d", parent_pid);
+
+	listen_pid = getenv("LISTEN_PID");
+	if (listen_pid != NULL && strcmp (listen_pid, pid_str) == 0) {
+		int r;
+
+		sprintf(pid_str, "%d", getpid());
+
+		r = setenv("LISTEN_PID", pid_str, 1);
+		if (r < 0)
+			warn("could not overwrite env variable LISTEN_PID: %m");
+	}
+}
+
```

Comment:
While I get that on paper it's safe in this specific case, I don't like the use of `sprintf` so please use `snprintf` or `asprintf` -- but more importantly I'd prefer `strtoll` for the pid equivalence check.